OpenRA.Mods.Common/Traits/Player/MissionObjectives.cs

```diff
@@ -172,34 +182,26 @@ void INotifyWinStateChanged.OnPlayerWon(Player player)
 						p.WinState = WinState.Won;
 						p.World.OnPlayerWinStateChanged(p);
 					}
-
-					if (Info.EarlyGameOver)
-						foreach (var p in enemies)
-							p.PlayerActor.Trait<MissionObjectives>().ForceDefeat(p);
 				}
 			}
 			else
 			{
 				player.WinState = WinState.Won;
 				player.World.OnPlayerWinStateChanged(player);
-
-				if (Info.EarlyGameOver)
-					foreach (var p in enemies)
-						p.PlayerActor.Trait<MissionObjectives>().ForceDefeat(p);
 			}
 
+			if (Info.EarlyGameOver)
```

Comment:
I don't understand what is going on with `WinStateCooperative` at all. Are you sure that it is safe to pull this out, which means it now runs if `Info.Cooperative` is true but at least one ally doesn't have `WinStateCooperative == WinState.Won`? It seems odd that we may be defeating all the enemies when some allies may not have yet won.

Similar question for OnPlayerLost below.

Reply:
Ah, that is a big derp from my side. Will revert that change.